backend/dataall/modules/loader.py

```diff
@@ -58,19 +67,35 @@ def depends_on() -> List[Type['ModuleInterface']]:
         return []
 
 
-def load_modules(modes: List[ImportMode]) -> None:
+def load_modules(modes: Set[ImportMode]) -> None:
     """
     Loads all modules from the config
     Loads only requested functionality (submodules) using the mode parameter
     """
+
+    to_load = _new_modules(modes)
+    if not to_load:
+        return
+
     in_config, inactive = _load_modules()
-    _check_loading_correct(in_config, modes)
-    _initialize_modules(modes)
+    _check_loading_correct(in_config, to_load)
+    _initialize_modules(to_load)
     _describe_loading(in_config, inactive)
 
     log.info("All modules have been imported")
 
 
+def _new_modules(modes: Set[ImportMode]):
+    """
+    Extracts only new modules to load. It's needed to avoid multiply loading
+    """
+    all_modes = _ACTIVE_MODES
+
+    to_load = modes - all_modes  # complement of set
+    all_modes |= modes
```

Comment:
It's needed for the tests. Some tests are testing the code that have `load_modules` (e.g. catalog_indexer), but tests also use `load_modules`. If we try to load modules twice `_check_loading_correct` will fail, so I added this line to avoid double loading.